services/app/src/reducers/Search.tsx

```diff
@@ -14,11 +16,21 @@ export const initialSearch: SearchState = {
 
 export function search(state: SearchState = initialSearch, action: Redux.Action): SearchState {
   switch (action.type) {
+    case 'CHANGE_SETTINGS':
+      // Clear results when invalidated.
+      return {...state, results: null};
+    case 'SEARCH_CHANGE_PARAMS':
+      // Update params and clear results
+      const changes = (action as SearchChangeParamsAction).params || {};
+      Object.keys(changes).forEach((key: string) => {
+        setStorageKeyValue(`search-${key}`, changes[key]);
```

Comment:
Seems like we're only using `search-language`... I'd be more comfortable if this just set that, until/unless we need the others for something.